tests/unit/test_charm.py

```diff
@@ -112,8 +105,18 @@ def test_given_upf_service_not_started_when_config_changed_then_service_started(
             ]
         )
 
-    @patch("charm.SnapCache")
-    def test_given_upf_services_started_when_remove_then_services_stopped(self, mock_snap_cache):
+    def test_given_upf_snap_uninstalled_when_remove_then_services_not_stopped(self):
```

Comment:
Do we actually need this test? It feels kinda strange to me to check if we're not stopping the services if they're not installed. I'd rather check that the services do not exist if the snap is not installed.

Reply:
This test was added in #32 since trying to stop a non-existent service causes the hook to fail. I am not sure how hard/easy it would be to check if the services are actually removed, as we are mocking the `SnapCache`.